src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -831,6 +887,34 @@ internal void EndSet(Type stackType)
                     _il.Emit(OpCodes.Stelem_Ref);
                 }
             }
+
+            private class PropertyAccessorInfo
+            {
+                public MethodInfo InterfaceGetMethod { get; private set; }
```

Comment:
InterfaceGetMethod can be made a read-only property, i.e.
 
  public MethodInfo InterfaceGetMethod { get; }

Same for InterfaceSetMethod.